registry/handlers/blobupload.go

```diff
@@ -241,10 +241,11 @@ func (buh *blobUploadHandler) PutBlobUploadComplete(w http.ResponseWriter, r *ht
 
 	desc, err := buh.Upload.Commit(buh, distribution.Descriptor{
 		Digest: dgst,
+		Size:   buh.State.Offset,
```

Comment:
The `State` will likely have the value at the start of the request. This value should be that from the result of `copyFullPayload`. Let's double check that this is correct.